BuildHelpers/Public/Get-BuildVariable.ps1

```diff
@@ -263,12 +264,31 @@ function Get-BuildVariable {
         $BuildNumber = 0
     }
 
-    [pscustomobject]@{
+    $PullRequest = switch ($BuildSystem)
+    {
+        'AppVeyor'        {if ($Environment.APPVEYOR_PULL_REQUEST_NUMBER) {"True"}; break}
+        'GitLab CI'       {if ($Environment.CI_MERGE_REQUEST_ID) {"True"}; break}
+        'Azure Pipelines' {if ($ENV:BUILD_REASON -eq "PullRequest") {"True"}; break}
+        'Bamboo'          {if ($Environment.bamboo.repository.pr.key) {"True"}; break}
+        'GoCD'            {if ($Environment.PR_TITLE) {"True"}; break}
+        'Travis CI'       {if ($ENV:TRAVIS_EVENT_TYPE -eq "pull_request") {"True"}; break}
+        'GitHub Actions'  {if ($ENV:GITHUB_EVENT_NAME -eq "pull_request") {"True"}; break}
+        #'Jenkins' {if ($Environment.CHANGE_ID) {"True"}} ???? is this correct?
+        #'Teamcity' {if ???????) {"True"}} who even knows
+    }
```

Comment:
Two questions!

* Can we add a default case that sets this to `$null`
* Can we add else statements to indicate `$false` when a build system is identified, but no pull request evidence is found (i.e. not in a PR, unlike default case of not detecting being in a PR, which is $null)
* Can we switch to boolean `$true`/`$false` rather than strings?

Sorry to be a pain, and sorry for such a long delay!

Reply:
Doing anything differently with the variable here (setting it to null or false) would be a change of my intended functionality.  Additionally, since these are going to be written to the environment as Environment variables, we'd lose the boolean-ness and they'd just end up as flat strings either way.

The intention was that the environment variable would only exist if it was a pull request, though I can see the benefit to knowing if you're definitively not in a PR build vs not being in a build environment where it can be detected.

Reply:
ok looked more closely at the code.  The usual entry point is going to be `Set-BuildEnvironment` which takes the values from `Get-BuildEnvironment` which is what calls this (`Get-BuildVariable`).

In the end it runs this line:
```powershell
$Output = New-Item -Path Env:\ -Name $prefixedVar -Value $BuildHelpersVariables[$VarName] -Force:$Force
```
If the value is `$true` that will set the Environment Variable to be 'True', if it's `$false` it will set it as 'False', but if it's `$null` this will throw an error.